Libraries/LibJS/Runtime/ObjectPrototype.cpp

```diff
@@ -138,4 +139,23 @@ JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::property_is_enumerable)
     return Value(property_descriptor.value().attributes.is_enumerable());
 }
 
+JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::is_prototype_of)
+{
+    auto object_argument = vm.argument(0);
+    if (!object_argument.is_object())
+        return Value(false);
+    auto* object = object_argument.to_object(global_object);
```

Comment:
Just for semantic correctness (there's no conversion going on, we know this is an object):

```suggestion
    auto* object = &object_argument.as_object();
```